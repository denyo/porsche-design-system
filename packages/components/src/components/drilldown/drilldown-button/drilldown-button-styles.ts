import {
  borderRadiusSmall,
  frostedGlassStyle,
  spacingFluidSmall,
  spacingStaticSmall,
  textMediumStyle,
} from '@porsche-design-system/styles';
import {
  addImportantToEachRule,
  colorSchemeStyles,
  getFocusJssStyle,
  getThemedColors,
  getTransition,
  hostHiddenStyles,
  hoverMediaQuery,
  prefersColorSchemeDarkMediaQuery,
  preventFoucOfNestedElementsStyles,
} from '../../../styles';
import { getCss, type Theme } from '../../../utils';

export const getComponentCss = (isActive: boolean, theme: Theme): string => {
  const { primaryColor, hoverColor } = getThemedColors(theme);
  const { primaryColor: primaryColorDark, hoverColor: hoverColorDark } = getThemedColors('dark');

  return getCss({
    '@global': {
      ':host': {
        display: 'grid',
        ...addImportantToEachRule({
          ...colorSchemeStyles,
          ...hostHiddenStyles,
        }),
      },
      ...preventFoucOfNestedElementsStyles,
      button: {
        all: 'unset',
        display: 'flex',
        gap: spacingStaticSmall,
        justifyContent: 'space-between',
        alignItems: 'center',
        padding: `calc(${spacingFluidSmall} + 2px) calc(${spacingFluidSmall} + 4px)`, // aligned with link-pure
        margin: `-2px calc(${spacingFluidSmall} * -1 - 4px)`, // aligned with link-pure
        borderRadius: borderRadiusSmall, // needed for focus outline and hover
        font: textMediumStyle.font,
        color: primaryColor,
        ...prefersColorSchemeDarkMediaQuery(theme, {
          color: primaryColorDark,
        }),
        cursor: isActive ? 'default' : 'pointer',
        transition: getTransition('background'),
        ...(isActive && {
          ...frostedGlassStyle,
          background: hoverColor,
          ...prefersColorSchemeDarkMediaQuery(theme, {
            background: hoverColorDark,
          }),
        }),
        ...hoverMediaQuery({
          '&:hover': {
            ...frostedGlassStyle,
            background: hoverColor,
            ...prefersColorSchemeDarkMediaQuery(theme, {
              background: hoverColorDark,
            }),
          },
        }),
        ...getFocusJssStyle(theme, { offset: '-2px' }),
      },
    },
  });
};
