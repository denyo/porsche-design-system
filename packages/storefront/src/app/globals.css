@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  code,
  pre {
    @apply pds-text-small;
    font-family: theme(fontFamily.code);
  }

  code.readonly {
    user-select: none;
  }

  :not(pre) > code {
    padding: 2px theme(spacing.static-sm);
    border-radius: theme(borderRadius.sm);
    background-color: theme(colors.background-code.light);
    color: theme(colors.primary.light);
  }

  pre {
    @apply pds-focus;
  }

  pre > * {
    margin-top: theme(spacing.static-sm);
    display: block;
    padding: theme(spacing.static-sm) theme(spacing.static-md);
    word-break: break-all;
    word-wrap: break-word;
    background-color: theme(colors.background-code.light);
    color: theme(colors.primary.light);
    border-radius: theme(borderRadius.md);
    overflow-x: auto;
    -webkit-overflow-scrolling: touch;
  }
}

@layer components {
  p-inline-notification {
    margin-top: theme(spacing.sm);
  }
}

/*TODO: Fix*/
@layer utilities {
  :where(h3, h4, h5, h6) + p-text-list {
    margin-top: theme(spacing.xs);
  }
}
