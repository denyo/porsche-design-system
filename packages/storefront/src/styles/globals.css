@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  body {
    background: theme(colors.background-surface.DEFAULT);
    color: theme(colors.primary.DEFAULT);
  }

  code {
    display: block;
    padding: theme(spacing.static-md);
    border-radius: theme(borderRadius.md);
    overflow-x: auto;
    background-color: theme(colors.background-code.DEFAULT);
    font-family: theme(fontFamily.code);
    font-weight: theme(fontWeight.regular);
    font-size: theme(fontSize.xs);
    line-height: 1.5;
    color: theme(colors.primary.DEFAULT);
  }

  :not(pre) > code {
    display: inline;
    padding: 2px 6px;
    border-radius: theme(borderRadius.sm);
    overflow-x: visible;
    font-weight: theme(fontWeight.bold);
    color: theme(colors.contrast-high.DEFAULT);
  }

  p-text-list :is(p-text-list, p-text) {
    margin: 0 !important;
  }

  p-inline-notification {
    margin-block: theme(spacing.sm);
  }

  .deprecated {
    color: theme(colors.notification-error.DEFAULT);
    text-transform: uppercase;
  }
}
