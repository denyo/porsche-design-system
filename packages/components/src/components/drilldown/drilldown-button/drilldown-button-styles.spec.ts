// import { validateCssAndMatchSnapshot } from '../../../../tests/unit/helpers';
import { getComponentCss } from './drilldown-button-styles';
import { expect } from '@jest/globals';

describe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([
    [true, 'light'],
    [false, 'light'],
    [false, 'dark'],
    [false, 'auto'],
  ])('should return correct css for isActive: %s and theme: %s', (...args) => {
    // validateCssAndMatchSnapshot(getComponentCss(...args));
    expect(getComponentCss(...args)).toMatchSnapshot();
  });
});
