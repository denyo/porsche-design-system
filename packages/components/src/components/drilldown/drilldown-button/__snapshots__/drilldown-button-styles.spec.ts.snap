// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for isActive: false and theme: auto 1`] = `
":host {
  display: grid;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
:not(:defined,[data-ssr]) {
  visibility: hidden;
}
button {
  all: unset;
  display: flex;
  gap: 8px;
  justify-content: space-between;
  align-items: center;
  padding: calc(clamp(8px, 0.5vw + 6px, 16px) + 2px) calc(clamp(8px, 0.5vw + 6px, 16px) + 4px);
  margin: -2px calc(clamp(8px, 0.5vw + 6px, 16px) * -1 - 4px);
  border-radius: 4px;
  font: normal normal 400 clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  color: #010205;
  cursor: pointer;
  transition: background var(--p-transition-duration, 0.25s) cubic-bezier(0.25,0.1,0.25,1);
}
button::-moz-focus-inner {
  border: 0;
}
button:focus {
  outline: 0;
}
button:focus-visible {
  outline: 2px solid #1A44EA;
  outline-offset: -2px;
}
@media (prefers-color-scheme: dark) {
  button {
    color: #FBFCFF;
  }
  button:focus-visible {
    outline-color: #1A44EA;
  }
}

@media(hover:hover) {
  button:hover {
    -webkit-backdrop-filter: blur(32px);
    backdrop-filter: blur(32px);
    background: rgba(148, 149, 152, .18);
  }
@media (prefers-color-scheme: dark) {
    button:hover {
      background: rgba(148, 149, 152, .18);
    }
}

}
"
`;

exports[`getComponentCss() should return correct css for isActive: false and theme: dark 1`] = `
":host {
  display: grid;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
:not(:defined,[data-ssr]) {
  visibility: hidden;
}
button {
  all: unset;
  display: flex;
  gap: 8px;
  justify-content: space-between;
  align-items: center;
  padding: calc(clamp(8px, 0.5vw + 6px, 16px) + 2px) calc(clamp(8px, 0.5vw + 6px, 16px) + 4px);
  margin: -2px calc(clamp(8px, 0.5vw + 6px, 16px) * -1 - 4px);
  border-radius: 4px;
  font: normal normal 400 clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  color: #FBFCFF;
  cursor: pointer;
  transition: background var(--p-transition-duration, 0.25s) cubic-bezier(0.25,0.1,0.25,1);
}
button::-moz-focus-inner {
  border: 0;
}
button:focus {
  outline: 0;
}
button:focus-visible {
  outline: 2px solid #1A44EA;
  outline-offset: -2px;
}
@media(hover:hover) {
  button:hover {
    -webkit-backdrop-filter: blur(32px);
    backdrop-filter: blur(32px);
    background: rgba(148, 149, 152, .18);
  }
}
"
`;

exports[`getComponentCss() should return correct css for isActive: false and theme: light 1`] = `
":host {
  display: grid;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
:not(:defined,[data-ssr]) {
  visibility: hidden;
}
button {
  all: unset;
  display: flex;
  gap: 8px;
  justify-content: space-between;
  align-items: center;
  padding: calc(clamp(8px, 0.5vw + 6px, 16px) + 2px) calc(clamp(8px, 0.5vw + 6px, 16px) + 4px);
  margin: -2px calc(clamp(8px, 0.5vw + 6px, 16px) * -1 - 4px);
  border-radius: 4px;
  font: normal normal 400 clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  color: #010205;
  cursor: pointer;
  transition: background var(--p-transition-duration, 0.25s) cubic-bezier(0.25,0.1,0.25,1);
}
button::-moz-focus-inner {
  border: 0;
}
button:focus {
  outline: 0;
}
button:focus-visible {
  outline: 2px solid #1A44EA;
  outline-offset: -2px;
}
@media(hover:hover) {
  button:hover {
    -webkit-backdrop-filter: blur(32px);
    backdrop-filter: blur(32px);
    background: rgba(148, 149, 152, .18);
  }
}
"
`;

exports[`getComponentCss() should return correct css for isActive: true and theme: light 1`] = `
":host {
  display: grid;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
:not(:defined,[data-ssr]) {
  visibility: hidden;
}
button {
  all: unset;
  display: flex;
  gap: 8px;
  justify-content: space-between;
  align-items: center;
  padding: calc(clamp(8px, 0.5vw + 6px, 16px) + 2px) calc(clamp(8px, 0.5vw + 6px, 16px) + 4px);
  margin: -2px calc(clamp(8px, 0.5vw + 6px, 16px) * -1 - 4px);
  border-radius: 4px;
  font: normal normal 400 clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  color: #010205;
  cursor: default;
  transition: background var(--p-transition-duration, 0.25s) cubic-bezier(0.25,0.1,0.25,1);
  -webkit-backdrop-filter: blur(32px);
  backdrop-filter: blur(32px);
  background: rgba(148, 149, 152, .18);
}
button::-moz-focus-inner {
  border: 0;
}
button:focus {
  outline: 0;
}
button:focus-visible {
  outline: 2px solid #1A44EA;
  outline-offset: -2px;
}
@media(hover:hover) {
  button:hover {
    -webkit-backdrop-filter: blur(32px);
    backdrop-filter: blur(32px);
    background: rgba(148, 149, 152, .18);
  }
}
"
`;
